sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessor.cs

```diff
@@ -146,19 +147,16 @@ internal EventProcessorStatus Status
 
                 if (_runningProcessorTask == null)
                 {
-                    try
+                    if (!ProcessorRunningGuard.Wait(100))
                     {
-                        if (!ProcessorRunningGuard.Wait(100))
-                        {
-                            return (_statusOverride ?? EventProcessorStatus.NotRunning);
-                        }
-
-                        statusOverride = _statusOverride;
-                    }
-                    finally
-                    {
-                        ProcessorRunningGuard.Release();
```

Comment:
This was the cause of the race condition.  We were releasing even if we did not acquire on L151.  This caused the release done in start/stop to throw.